### ISSUE TITLE
git worktree changes paths after add

### ISSUE THREAD

**Issue by therealprof** [git/git-scm.com#650]
Every know and then after adding a new worktree git worktree list will an incorrect path for a previously added worktree:

State before:
```
# git worktree list
/Users/user/Desktop/Source/Product          2991e32 [master]
/Users/user/Desktop/Source/Product-2.1      3fc1082 [releases/master-2.1]
/Users/user/Desktop/Source/Product-2.2      3ccc3fe [releases/master-2.2]
/Users/user/Desktop/Source/Product-2.3      4044881 [releases/master-2.3]
/Users/user/Desktop/Source/Product-2.3-S11  4a67d36 [customer/S11]
/Users/user/Desktop/Source/Product-2.2-S22  2d2503a [customer/S22]
```

State after adding a new worktree:
```
# git worktree list
/Users/user/Desktop/Source/Product          2991e32 [master]
/Users/user/Desktop/Source/Product-2.1      3fc1082 [releases/master-2.1]
/Users/user/Desktop/Source/Product          3ccc3fe [releases/master-2.2]
/Users/user/Desktop/Source/Product-2.3      4044881 [releases/master-2.3]
/Users/user/Desktop/Source/Product-2.3-S11  4a67d36 [customer/S11]
/Users/user/Desktop/Source/Product-2.3-S33  174575e [customer/S33]
/Users/user/Desktop/Source/Product-2.2-S22  2d2503a [customer/S22]
```

3ccc3fe is now (incorrectly) claimed to be represented in /Users/user/Desktop/Source/Product.

**Comment by terminalmage** [comment]
I noticed this too after upgrading to 2.7.0. This is because $GIT_DIR/worktrees/``worktree_name``/gitdir is for some reason being rewritten to merely say ``.git`` instead of ``/path/to/worktree/root/.git``.

**Comment by terminalmage** [comment]
I don't know yet what is triggering this, it doesn't seem to be triggered by the process of adding a worktree, so my guess is that it might have something to do with running ``git worktree list`` since I run that much more often. But that is just a guess and nothing more.

**Comment by jnavila** [comment]
This isn't the right place to report a bug in Git. This projet is dedicated to the git-scm.com site, not the Git project itself.

Bug reports in Git should be reported to the git mailing list: git@vger.kernel.org

**Comment by therealprof** [comment]
Sorry, my bad. Misread the bug reporting guidelines on the homepage.

**Comment by terminalmage** [comment]
As did I. Apologies.

**Comment by therealprof** [comment]
Hah, I figured it out. git reset --hard in one of the worktrees will mess up the directory. I'm going to close this and point it out at the mailing list.

**Comment by terminalmage** [comment]
@therealprof awesome, thanks for narrowing it down. I think they have a handle on it and a fix coming soon in 2.7.1.

**Comment by ewirch** [comment]
Any news on this? I'm running 2.9.3 and still experiencing this bug.
